src/components/Post/Comment/EditableCommentContent.tsx

```diff
@@ -285,52 +284,58 @@ const EditableCommentContent: FC<IEditableCommentContentProps> = (props) => {
-			commentId,
-			postId: ((comment.post_index || comment.post_index === 0)? comment.post_index: props.postId),
-			postType: comment.post_type || props.proposalType,
-			trackNumber: track_number
-		});
-
-		if (deleteCommentError || !data) {
-			console.error('Error deleting comment: ', deleteCommentError);
-			queueNotification({
-				header: 'Error!',
-				message: deleteCommentError || 'There was an error in deleting your comment.',
-				status: NotificationStatus.ERROR
-			});
-		}
-
-		if(data) {
-			const keys = Object.keys(comments);
-			setComments((prev) => {
-				const comments: any = Object.assign({}, prev);
-				for(const key of keys ){
-					if (prev?.[key]) {
-						comments[key]  = prev[key].filter((comment) => comment.id !== commentId);
-					}
+	const removeCommentContent = () => {
+		const keys = Object.keys(comments);
+		setComments((prev) => {
+			const comments: any = Object.assign({}, prev);
+			for(const key of keys ){
+				if (prev?.[key]) {
+					comments[key]  = prev[key].filter((comment) => comment.id !== commentId);
 				}
-				return comments;
-			});
-			setTimelines((prev) => {
-				return [...prev.map((timeline) => {
-					if (timeline.index === `${postIndex}` && timeline.type === getSubsquidLikeProposalType(postType)) {
-						return {
-							...timeline,
-							commentsCount: (timeline.commentsCount > 0? timeline.commentsCount - 1: 0)
-						};
-					}
+			}
+			return comments;
+		});
+		setTimelines((prev) => {
+			return [...prev.map((timeline) => {
+				if (timeline.index === `${postIndex}` && timeline.type === getSubsquidLikeProposalType(postType)) {
 					return {
-						...timeline
+						...timeline,
+						commentsCount: (timeline.commentsCount > 0? timeline.commentsCount - 1: 0)
 					};
-				})];
-			});
-			queueNotification({
-				header: 'Success!',
-				message: 'Your comment was deleted.',
-				status: NotificationStatus.SUCCESS
+				}
+				return {
+					...timeline
+				};
+			})];
+		});
+		queueNotification({
+			header: 'Success!',
+			message: `${allowed_roles?.includes('moderator') && userId! == id? 'The' : 'Your'} comment was deleted.`,
+			status: NotificationStatus.SUCCESS
+		});
+	};
+
+	const deleteComment = async () => {
+		if (!allowed_roles?.includes('moderator')) {
+			const { data, error: deleteCommentError } = await nextApiClientFetch<MessageType>('api/v1/auth/actions/deleteComment', {
+				commentId,
+				postId: ((comment.post_index || comment.post_index === 0) ? comment.post_index : props.postId),
+				postType: comment.post_type || props.proposalType
 			});
+			if (deleteCommentError || !data) {
+				console.error('Error deleting comment: ', deleteCommentError);
+				queueNotification({
+					header: 'Error!',
+					message: deleteCommentError || 'There was an error in deleting your comment.',
+					status: NotificationStatus.ERROR
+				});
+			}
+			if (data){
+				removeCommentContent();
+			}
+		}
+		else{
+			removeCommentContent();
```

Comment:
if a user is moderator and clicks on delete (when deleting his own comment), his comment is not deleted herebut only removed from state

Reply:
this else clause needs to be removed

Reply:
done, pls check